Progress file errors spam output when file is deleted mid-run

## Problem

During a shepherd run, repeated "[ERROR] No progress file found for task X" messages appear in the output despite the `_progress_initialized` guard in `context.py`.

## Observed Output

```
[20:55:18] [INFO] Completion phase finished, re-validating
[20:55:19] [ERROR] No progress file found for task c3f9ae8
[20:55:19] Run 'report-milestone.sh started --task-id c3f9ae8 --issue N' first
[20:55:19] [OK] Builder phase complete - PR #381 created (1027s)
[20:55:19] [ERROR] No progress file found for task c3f9ae8
[20:55:19] Run 'report-milestone.sh started --task-id c3f9ae8 --issue N' first
```

No progress file `shepherd-c3f9ae8.json` exists in `.loom/progress/` after the run.

## Root Cause Analysis

The `_progress_initialized` guard in `context.py:287` silently skips non-"started" milestone events when initialization fails:

```python
if event != "started" and not self._progress_initialized:
    return False
```

However, the errors appear to come from one of two scenarios:

**Scenario A: File deleted mid-run by concurrent shepherd**
If another shepherd starts for the same issue, `_cleanup_stale_progress_for_issue()` in `__post_init__` deletes progress files for the same issue with different task IDs. This removes the running shepherd's file. Since `_progress_initialized` was already set to True (the "started" event succeeded initially), subsequent calls pass through to `report_milestone()` which discovers the file is gone and logs the error.

**Scenario B: Components calling the milestone API directly**
`validate_phase.py:_report_milestone()` calls `report-milestone.sh` as a subprocess (line 138), bypassing `context.py`'s guard. While `capture_output=True` hides its output, there may be other code paths that invoke the milestone API without going through the context.

## Fix Options

1. **Handle "file disappeared" in context.py**: After `_report()` returns False for a non-"started" event, set `_progress_initialized = False` to re-engage the silent suppression:
   ```python
   result = _report(...)
   if not result and event != "started":
       self._progress_initialized = False
   return result
   ```

2. **Lock the progress file**: Use file locking to prevent concurrent deletion. This is heavier but prevents the root cause.

3. **Make `_cleanup_stale_progress_for_issue()` check for active processes**: Before deleting a progress file, check if the task is still running (e.g., via a PID recorded in the progress file).

4. **Handle missing file gracefully in `milestones.py`**: Instead of logging ERROR for "file not found", downgrade to DEBUG for non-critical events (heartbeat, phase_entered) and only log ERROR for critical events (completed, blocked, error).

## Files

- `loom-tools/src/loom_tools/shepherd/context.py` (lines 284-319 — `report_milestone`)
- `loom-tools/src/loom_tools/milestones.py` (lines 233-239 — missing file error)
- `loom-tools/src/loom_tools/validate_phase.py` (lines 129-147 — `_report_milestone`)

--- a/loom-tools/src/loom_tools/milestones.py
+++ b/loom-tools/src/loom_tools/milestones.py
@@ -231,11 +231,10 @@ def report_milestone(
             return True
 
         # All other events require an existing progress file.
+        # The file may have been deleted by a concurrent shepherd cleaning
+        # up stale progress for the same issue — use warning, not error.
         if not progress_file.is_file():
-            log_error(f"No progress file found for task {task_id}")
-            log_error(
-                f"Run 'report-milestone.sh started --task-id {task_id} --issue N' first"
-            )
+            log_warning(f"No progress file found for task {task_id} (may have been cleaned up by another shepherd)")
             return False
 
         raw = read_json_file(progress_file)

--- a/loom-tools/src/loom_tools/shepherd/context.py
+++ b/loom-tools/src/loom_tools/shepherd/context.py
@@ -467,6 +467,17 @@ class ShepherdContext:
                         "Subsequent milestones will be skipped.",
                         self.config.task_id,
                     )
+            elif not result:
+                # Progress file may have been deleted mid-run by a concurrent
+                # shepherd cleaning up stale files for the same issue.
+                # Re-engage the silent suppression guard so subsequent
+                # milestone calls don't produce repeated error messages.
+                self._progress_initialized = False
+                logger.debug(
+                    "Progress file for task %s appears to have been removed. "
+                    "Disabling further milestone reporting.",
+                    self.config.task_id,
+                )
 
             return result
         except Exception as exc:

